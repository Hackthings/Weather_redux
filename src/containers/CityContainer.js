import React, { Component } from 'react';

const url = "http://api.wunderground.com/api/4af47e77859b009e/conditions/q/CA/";


class CityContainer extends Component {

    constructor(props) {
        super(props);

        this.state = {
            data: null
        }
    }

    componentDidMount() {
       this.requestInitialData(this.props.city);
    }

    async requestInitialData(city) {
        const requestUrl = url + city + '.json';
        try {
            const response = await fetch(requestUrl);
            const parseResponse = await response.json();
            console.log(parseResponse.current_observation)
            this.setState({
                data: parseResponse.current_observation
            })
        }
        catch (error) {
            console.log('there was an error')
        }
    }

    render() {
        const { data } = this.state;
        
        return (
            <div>
                {!data && (<span>loading...</span>)}
                {data && (<span>{data.icon}</span>)}
            </div>
        )
    }


}

export default CityContainer;