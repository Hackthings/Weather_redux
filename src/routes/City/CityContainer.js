import React, { Component } from 'react';
import City from './City';

const url = "http://api.wunderground.com/api/4af47e77859b009e/conditions/q/";
const stateAbbreviation = 'CA';


class CityContainer extends Component {

    constructor(props) {
        super(props);

        this.state = {
            data: null,
            isRequestingInitialData: true
        }
    }

    componentDidMount() {
        const cityName = this.props.match.params.name;
        this.requestInitialData(cityName);
    }

    async requestInitialData(city) {
        const requestUrl = url + stateAbbreviation + '/' + city + '.json';
        this.setState({
            isRequestingInitialData: true
        });
        try {
            const response = await fetch(requestUrl);
            const parseResponse = await response.json();
            const parsed = this.parseResponse(parseResponse)

            console.log(parseResponse.current_observation)
            this.setState({
                ...parsed,
                isRequestingInitialData: false
            })
        }
        catch (error) {
            console.log('there was an error')
        }
    }

    parseResponse(response) {
        const {
            current_observation: {
                display_location: {
                    city,
                    country,
                    state_name
                },
                feelslike_c,
                icon,
                icon_url,
                local_time_rfc822
            }
        } = response;
        return {
            city,
            country,
            stateName: state_name,
            temperature: feelslike_c,
            icon,
            iconUrl: icon_url,
            localTime: local_time_rfc822
        }

    }

    render() {
        const {
            isRequestingInitialData,
            city,
            country,
            stateName,
            temperature,
            icon,
            iconUrl,
            localTime
        } = this.state;

        return (
            <City
                isRequestingInitialData={isRequestingInitialData}
                city={city}
                country={country}
                stateName={stateName}
                temperature={temperature}
                icon={icon}
                iconUrl={iconUrl}
                localTime={localTime}
            />
        )
    }


}

export default CityContainer;