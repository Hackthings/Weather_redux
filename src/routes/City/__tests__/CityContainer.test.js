import React from 'react';
import CityContainer from '../CityContainer';
import { shallow, mount } from 'enzyme';

describe('<CityContainer/>', () => {
    describe('parseResponse', () => {

        const props = {
            match: {
                params: {
                    name: 'test'
                }
            }
        };
        const wrapper = shallow(<CityContainer {...props} />);
        const { parseResponse } = wrapper.instance();

        test('empty params', () => {
            expect(parseResponse).toThrow();
        });

        test('correct params', () => {
            expect(parseResponse({
                current_observation: {
                    display_location: {
                        city: 'test1',
                        country: 'test2',
                        state_name: 'test3'
                    },
                    feelslike_c: 'test4',
                    icon: 'test5',
                    icon_url: 'test6',
                    local_time_rfc822: 'test7'
                }
            })).toMatchSnapshot();
        });

    })

})
