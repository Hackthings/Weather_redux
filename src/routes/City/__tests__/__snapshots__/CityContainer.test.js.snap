// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityContainer/> parseResponse correct params 1`] = `
Object {
  "city": "test1",
  "country": "test2",
  "icon": "test5",
  "iconUrl": "test6",
  "localTime": "test7",
  "stateName": "test3",
  "temperature": "test4",
}
`;
